File: mobile/src/main/java/com/novoda/materialised/hackernews/stories/IntentNavigator.java
package com.novoda.materialised.hackernews.stories;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;

import com.novoda.materialised.hackernews.navigator.Navigator;

import org.jetbrains.annotations.NotNull;

final class IntentNavigator implements Navigator {
    private final Context context;

    IntentNavigator(Context context) {
        this.context = context;
    }

    @Override
    public void navigateTo(@NotNull String uri) {
        context.startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(uri)));
    }
}

File: mobile/src/main/java/com/novoda/materialised/hackernews/stories/IntentNavigator.kt
package com.novoda.materialised.hackernews.stories

import android.content.Context
import android.content.Intent
import android.net.Uri

import com.novoda.materialised.hackernews.navigator.Navigator

internal class IntentNavigator(private val context: Context) : Navigator {

    override fun navigateTo(uri: String) = context.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(uri)))
}
